skmultiflow/classification/meta/adaptive_random_forests.py

```diff
@@ -0,0 +1,396 @@
+
+INSTANCE_WEIGHT = np.array([1.0])
+FEATURE_MODE_M = ''
+FEATURE_MODE_SQRT = 'sqrt'
+FEATURE_MODE_SQRT_INV = 'sqrt_inv'
+FEATURE_MODE_PERCENTAGE = 'percentage'
+
+
+class AdaptiveRandomForest(BaseClassifier):
+    """Adaptive Random Forest (ARF).
+
+        Parameters
+        ----------
+        nb_ensemble: int (default=10)
+            Number of trees oin the ensemble
+        feature_mode: string (default='sqrt')
+            / How the subspace size value is interpreted to calculate the number of attributes.
+            / 'sqrt' - use the square root value ```sqrt(M)+1```
+            / 'sqrt_inv' - ```M-(sqrt(M)+1)```
+            / 'percentage' - Use as percentage
+            / '' - use subspace value as the number of attributes to use.
+        subspace_size: int or float (default=2)
+            Number of attributes per subset for each node split.
+        disable_background_learner: bool (default=False)
+            Background learner option.
+        disable_drift_detection: bool (default=False)
+            Drift detection option.
+        disable_weighted_vote: bool (default=False)
+            Weighted vote option.
+        lambda_value: int (default=6)
+            The lambda value for bagging (lambda=6 corresponds to Leverage Bagging).
+        evaluator_method_metric: string (default='acc')
+            / Metric used to track tree performance.
+            / 'acc' - Accuracy
+        drift_detection_method: BaseDriftDetector default(ADWIN(0.001))
+            Drift Detection method
+        warning_detection_method: BaseDriftDetector default(ADWIN(0.01))
+            Warning Detection method
+        nominal_attributes: list, optional
+            List of Nominal attributes. If emtpy, then assume that all attributes are numerical.
+
+
+        Notes
+        -----
+        The 3 most important aspects of Adaptive Random Forest [1]_ are:
+        (1) inducing diversity through re-sampling;
+        (2) inducing diversity through randomly selecting subsets of features for node splits (see
+        skmultiflow.classification.trees.arf_hoeffding_tree);
+        (3) drift detectors per base tree, which cause selective resets in response to drifts.
+        It also allows training background trees, which start training if a warning is detected and replace the active
+        tree if the warning escalates to a drift.
+
+        References
+        ----------
+        .. [1] Heitor Murilo Gomes, Albert Bifet, Jesse Read, Jean Paul Barddal, Fabricio Enembreck,
+           Bernhard Pfharinger, Geoff Holmes, Talel Abdessalem.
+           Adaptive random forests for evolving data stream classification.
+           In Machine Learning, DOI: 10.1007/s10994-017-5642-8, Springer, 2017.
+
+    """
+
+    def __init__(self,
+                 nb_ensemble=10,
+                 feature_mode='sqrt',
+                 subspace_size=2,
+                 disable_background_learner=False,
```

Comment:
Maybe these disable_* flags are not necessary for the drift detection part, perhaps this could be controlled by setting the drift_detection_method or `warning_detection_method` to None.

Reply:
Nice, small signatures for the win.

- [x] Removed disable_background_learner and disable_drift_detection